app/controllers/albums_controller.rb

```diff
@@ -0,0 +1,62 @@
+class AlbumsController < ApplicationController
+  before_action :get_album, only: [:edit, :update, :destroy]
+  def get_album
+    @album = Album.find(params[:id])
+  end
+
+  def index
+    @albums = Album.all
+  end
+
+  def new
+    @url = "create"
+    @legend = "New Album"
+    @album = Album.new
+  end
+
+  def create
+    @album = Album.create(album_params[:album])
+    if @album.save
+      redirect_to albums_path
+    else
+      render :new
+    end
+  end
+
+  def show
+    @album = Album.find(params[:id])
+  end
+
+  def edit
+    @album = Album.find(params[:id])
+    @legend = "Edit Album"
+  end
+
+  def update
+    @album = Album.update(params[:id], album_params[:album])
+      if @album.save
+        redirect_to album_path(params[:id])
+      else
+        render :edit
+      end
+  end
+
+  def destroy
+    Album.destroy(params[:id])
+    redirect_to albums_path
+  end
+
+  def upvote
+    @album = Album.find(params[:id])
+    @album.rank += 1
```

Comment:
This works beautifully and is generally my go to as it works across many different programming languages. Another way to accomplish the same thing in Ruby is the method increment. Yay options!